document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/MongoQueryExecutor.java

```diff
@@ -130,7 +130,8 @@ public MongoCursor<BasicDBObject> aggregate(final Query originalQuery) {
     logPipeline(pipeline);
 
     try {
-      final AggregateIterable<BasicDBObject> iterable = collection.aggregate(pipeline);
+      final AggregateIterable<BasicDBObject> iterable =
+          collection.aggregate(pipeline).allowDiskUse(true);
```

Comment:
In what scenarios it will use disk ? Depends on size of data or by default it uses disk  ?

Reply:
> Use [allowDiskUse()](https://www.mongodb.com/docs/manual/reference/method/cursor.allowDiskUse/?_ga=2.154934904.1773902492.1716202636-852641550.1716202636#mongodb-method-cursor.allowDiskUse) to either allow or prohibit writing temporary files on disk when a pipeline stage exceeds the 100 megabyte limit. Starting in MongoDB 6.0, operations that require greater than 100 megabytes of memory automatically write data to temporary files by default.

Ref.: https://www.mongodb.com/docs/manual/reference/method/cursor.allowDiskUse/?_ga=2.154934904.1773902492.1716202636-852641550.1716202636#mongodb-method-cursor.allowDiskUse